server/rest/src/main/java/org/hawkular/apm/server/rest/AnalyticsHandler.java

```diff
@@ -500,6 +501,108 @@ public void getTraceCompletionFaultCount(
+            @ApiParam(required = false,
+            value = "traces after this time,"
+                    + " millisecond since epoch") @DefaultValue("0") @QueryParam("startTime") long startTime,
+            @ApiParam(required = false,
+                    value = "traces before this time, "
+                            + "millisecond since epoch") @DefaultValue("0") @QueryParam("endTime") long endTime,
+            @ApiParam(required = false,
+                            value = "host name") @QueryParam("hostName") String hostName,
+            @ApiParam(required = false,
+                            value = "principal") @QueryParam("principal") String principal,
+            @ApiParam(required = false,
+                            value = "traces with these properties, defined as a comma "
+                                    + "separated list of name|value "
+                                    + "pairs") @DefaultValue("") @QueryParam("properties") String properties,
+            @ApiParam(required = false,
+                                    value = "faults") @QueryParam("faults") String faults) {
+
+        try {
+            Criteria criteria = new Criteria();
+            criteria.setBusinessTransaction(businessTransaction);
+            criteria.setStartTime(startTime);
+            criteria.setEndTime(endTime);
+            criteria.setHostName(hostName);
+            criteria.setPrincipal(principal);
+
+            RESTServiceUtil.decodeProperties(criteria.getProperties(), properties);
+
+            RESTServiceUtil.decodeFaults(criteria.getFaults(), faults);
+
+            log.tracef("Get trace completion times for criteria (GET) [%s]",
+                    criteria);
+
+            List<CompletionTime> times = analyticsService.getTraceCompletionTimes(
+                    securityProvider.validate(tenantId, context.getUserPrincipal().getName()), criteria);
+
+            log.tracef("Got trace completion times for criteria (GET) [%s] = %s",
+                    criteria, times);
+
+            response.resume(Response.status(Response.Status.OK).entity(times).type(APPLICATION_JSON_TYPE)
+                    .build());
+
+        } catch (Throwable e) {
+            log.debug(e.getMessage(), e);
+            Map<String, String> errors = new HashMap<String, String>();
+            errors.put("errorMsg", "Internal Error: " + e.getMessage());
+            response.resume(Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+                    .entity(errors).type(APPLICATION_JSON_TYPE).build());
+        }
+
+    }
+
+    @POST
+    @Path("trace/completion/times")
+    @Produces(APPLICATION_JSON)
+    @ApiOperation(
+            value = "Get the trace completion times associated with criteria",
+            response = List.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Success"),
+            @ApiResponse(code = 500, message = "Internal server error") })
+    public void getTraceCompletionTimes(
```

Comment:
Yes it is doing the same - however for convenience I provide two ways to invoke the operation, one using GET and passing the parameters, but from a Javascript app (i.e. the UI) it is better to POST an object representation of the Criteria being managed in the javascript code.